compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -1504,13 +1504,18 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(using Context) {
       assert(tree.hasType, tree)
       val qual1 = typed(tree.qualifier, shallowSelectionProto(tree.name, pt, this))
       val resNoReduce = untpd.cpy.Select(tree)(qual1, tree.name).withType(tree.typeOpt)
-      val resMaybeReduced = constToLiteral(reducer.reduceProjection(resNoReduce))
-      if (resNoReduce ne resMaybeReduced)
-        typed(resMaybeReduced, pt) // redo typecheck if reduction changed something
+      val reducedProjection = reducer.reduceProjection(resNoReduce)
+      if (reducedProjection.isType)
+        //in case the projection leads to a typed tree, then there is nothing to reduce
```

Comment:
Technically we could dealias a type selection which is a form of reduction, it's just not useful compared to reducing terms so we don't do it.